### PR TITLE
Added support for logging unhandled promise rejections

### PR DESCRIPTION
**Pull request by aaronpowell**
- New global handled for unhandledrejection, similar to onerror
- New config property to test for

Addresses #1013

Replaced #1069

### CODE REVIEW

File: extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts [hunk 1]
```diff
@@ -576,6 +580,28 @@ export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IApp
             this.config.autoExceptionInstrumented = true;
         }
 
+        if (this.config.disableExceptionTracking === false &&
+            !this.config.autoUnhandledPromiseInstrumented) {
+            // We want to enable exception auto collection and it has not been done so yet
+            const onunhandledrejection = "onunhandledrejection";
+            const originalOnUnhandledRejection = window[onunhandledrejection];
```
Comment (MSNev) [inline extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts]: As part of my unload PR, I've added a getWindow() helper method which you will need to use as the SDK can be run in an environment without window, document, navigator etc defined. And this causes a ReferenceError exception to be thrown when accessing window in one of these environments.

Hopefully, the react issue and my PR can be checked in shortly so we can pick up your change as well.

Comment (MSNev) [reply extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts]: My Unload and EnvUtils (hasWindow(), getWindow()) is now checked in and I've disabled the failing reactivenative tests

Comment (aaronpowell) [reply extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts]: Shall I merge in that branch and use the `getWindow` function?

Comment (MSNev) [reply extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts]: If you rebase to master you will get the methods and then yes, if you use getWindow() (Checking that it doesn't return null and/or use hasWindow() before hooking the events) then this looks good.

Comment (aaronpowell) [reply extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts]: Updated the PR with a rebase of `master` and using the new method (see https://github.com/microsoft/ApplicationInsights-JS/pull/1070/commits/4788af78d584c90087ab1d6ad910702af2b2d344)

File: extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts [hunk 2]
```diff
@@ -585,6 +589,28 @@ export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IApp
             this.config.autoExceptionInstrumented = true;
         }
 
+        if (this.config.disableExceptionTracking === false &&
+            !this.config.autoUnhandledPromiseInstrumented) {
+            // We want to enable exception auto collection and it has not been done so yet
+            const onunhandledrejection = "onunhandledrejection";
+            const originalOnUnhandledRejection = _window[onunhandledrejection];
+            _window.onunhandledrejection = (error: PromiseRejectionEvent) => {
```
Comment (MSNev) [inline extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts]: _window may be null (in an electron app), please add a "&& _window" check to the if statement on lines 592-593

Comment (aaronpowell) [reply extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts]: Done, and I also rebased `master`.

**Pull request merged**